test/controllers/predicates/AuthorisedActionSpec.scala

```diff
@@ -407,6 +407,23 @@ class AuthorisedActionSpec extends TestUtils {
           status(result) mustBe UNAUTHORIZED
         }
       }
+
+      "results in a non-Auth related Exception to be returned for Primary and Secondary Agent check" in {
```

Comment:
please split this test for primary and secondary.
`  "results in a non-Auth related Exception to be returned for Primary and Secondary Agent check" in {

        object NonAuthException extends Exception("Non-authentication related exception")
      "results in a non-Auth related Exception to be returned as Internal Server Error for Primary Agent and secondary agent, check" in {

        lazy val result = {
          (mockAuthConnector.authorise(_: Predicate, _: Retrieval[_])(_: HeaderCarrier, _: ExecutionContext))
            .expects(*, Retrievals.allEnrolments, *, *)
            .returning(Future.failed(NonAuthException))

          mockAuthReturnException(new Exception("Bang"))
          auth.agentAuthentication(block, "1234567890")(fakeRequest, emptyHeaderCarrier)
        }

        status(result) mustBe INTERNAL_SERVER_ERROR

      }

      "results in a non-Auth related Exception to be returned as Internal Server Error for secondary agent, check" in {
        lazy val result = {
          (() => mockedAppConfig.emaSupportingAgentsEnabled).expects().returning(true)

          mockAuthReturnException(InsufficientEnrolments())
          mockAuthReturnException(new Exception("Bang"))
          auth.agentAuthentication(block, "1234567890")(fakeRequest, emptyHeaderCarrier)
        }

        status(result) mustBe INTERNAL_SERVER_ERROR

      }`